### DISCUSSION TITLE
Issue with switch

### DISCUSSION THREAD

**Discussion by fmadrid-chwy** [mockoon/mockoon#554] [Help] [answered]
In my app, we have a "path" key where we calculate a concatenated path of the parent's above and the child when it's created. I'm currently trying to find a way to reproduce this behavior based on the parentId in the body. When I try to use the "switch" helper, it will insert the whitespaces and switch output at the line in which it was executed. Its valid JSON (I think), but as the N cases increase, the body will become longer and longer. It's not a problem for machines, but its' hard for my colleagues and I to work with.

Ask: How can I remove the whitespaces around the response? Is there a better way to handle this? I wouldn't mind being able to calculate the path ahead of time and insert it, but I find I cannot create handlebar partials or helpers myself.

Input: `curl -X POST -H'content-type:application/json' -d '{"identifier": "snickers", "parentId": "10000000066"}' http://localhost:8085/master-taxonomy/v1/categories`

template:
```
{{!
{
  "identifier": "string",
  "displayName": "string",
  "longDescription": "string",
  "parentId": "string",
  "sequence": 0,
  "shortDescription": "string"
}
}}
{
  "categoryId": "{{ int }}",
  "created": "{{ now }}",
  "displayName": "{{ body 'displayName' }}",
  "identifier": "{{ body 'identifier' }}",
  "lastUpdated": "{{ now }}",
  "longDescription": "{{ body 'longDescription' }}",
  "path": {{# switch (body 'parentId') }}
    {{# case '10000000064' }}"{{ concat 'ROOT|' (body 'identifier') }}"{{/ case }}
    {{# case '10000000080' }}"{{ concat 'ROOT|m-dog|m-dog-food|' (body 'identifier') }}"{{/ case }}
    {{# case '10000000066' }}"{{ concat 'ROOT|m-dog|' (body 'identifier') }}"{{/ case }}
  {{/ switch }},
  "sequence": {{ body 'sequence' }},
  "shortDescription": "{{ body 'shortDescription' }}",
  "status": "ACTIVE"
}
```

Result:
```{
  "categoryId": "78401",
  "created": "2021-08-24T14:46:44.394-05:00",
  "displayName": "",
  "identifier": "snickers",
  "lastUpdated": "2021-08-24T14:46:44.394-05:00",
  "longDescription": "",
  "path":


    "ROOT|m-dog:snickers"
  ,
  "sequence": ,
  "shortDescription": "",
  "status": "ACTIVE"
}
```

**Comment by 255kb** [comment, marked as answer]
You could try to enclose the whole switch in double quotes instead of the case content. But I cannot test this right now and I am not sure it will work.
Another solution would be to use the [setVar](https://mockoon.com/docs/latest/templating/mockoon-helpers/#setvar) helper inside the cases and move the switch outside the content. Something like this (pseudo code):

```
{{#switch}}
{{#case}}{{setVar 'myVar' (concat ...)}}{{/case}}
{{/switch}}

{
  "property": "{{myVar}}"
}
```
